src/terminal/adapter/terminalOutput.cpp

```diff
@@ -136,37 +128,35 @@ bool TerminalOutput::DesignateCharset(const wchar_t wchNewCharset)
 // - <none>
 // Return Value:
 // - True if the current charset is not USASCII
-bool TerminalOutput::NeedToTranslate() const
+bool TerminalOutput::NeedToTranslate() const noexcept
 {
-    return _wchCurrentCharset != DispatchTypes::VTCharacterSets::USASCII;
+    return _currentCharset != DispatchTypes::VTCharacterSets::USASCII;
 }
 
-const wchar_t* TerminalOutput::_GetTranslationTable() const
+const std::wstring_view TerminalOutput::_GetTranslationTable() const noexcept
 {
-    const wchar_t* pwchTranslation = nullptr;
-    switch (_wchCurrentCharset)
+    switch (_currentCharset)
     {
     case DispatchTypes::VTCharacterSets::DEC_LineDrawing:
-        pwchTranslation = TerminalOutput::s_rgDECSpecialGraphicsTranslations;
-        break;
+        return { s_decSpecialGraphicsTranslations.data(), s_decSpecialGraphicsTranslations.size() };
     }
-    return pwchTranslation;
+    return {};
 }
 
-wchar_t TerminalOutput::TranslateKey(const wchar_t wch) const
+wchar_t TerminalOutput::TranslateKey(const wchar_t wch) const noexcept
 {
     wchar_t wchFound = wch;
-    if (_wchCurrentCharset == DispatchTypes::VTCharacterSets::USASCII ||
+    if (_currentCharset == DispatchTypes::VTCharacterSets::USASCII ||
         wch < '\x5f' || wch > '\x7f') // filter out the region we know is unchanged
     {
         ; // do nothing, these are the same as default.
     }
     else
     {
-        const wchar_t* pwchTranslationTable = _GetTranslationTable();
-        if (pwchTranslationTable != nullptr)
+        const auto translationTable = _GetTranslationTable();
+        if (!translationTable.empty())
         {
-            wchFound = (pwchTranslationTable[wch - '\x20']);
+            wchFound = translationTable.at(wch - '\x20');

```

Comment:
I dunno, the checked `at` here seems .. ... oh, seems like maybe it could be totally optimized out by a conforming optimizer? I dunno.

Reply:
What are you asking me to do here?

Reply:
We spoke verbally: just resolving this.